pkg/reconciler/trustedresources/sign.go

```diff
@@ -0,0 +1,89 @@
+/*
+Copyright 2022 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package trustedresources
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/sigstore/cosign/pkg/cosign"
```

Comment:
I'd drop the signing from this PR, since the pipeline controller will primarily be interested in verification only. (signing will likely be more relevant for CLI / CLI plugin).

In generaly, I'd avoid taking on the cosign dependency unless it's absolutely necessary - as you can see from the vendor directory, there's a ton of dependencies.

Reply:
I'm remembering now - you're using this for testing.

It looks like you're using cosign mostly for the key generation - you can drop this in favor of generating + loading the key directly via one of the load functions in https://pkg.go.dev/github.com/sigstore/sigstore/pkg/signature

e.g. you could replace most of this with something like:

```go
key, err := ecdsa.GenerateKey(elliptic.P256(), rand.Reader)
if err ...
sv, err := signature.LoadECDSASignerVerifier(key, crypto.SHA256)
if err ...
return sv
```

Reply:
if/when you need the multi-algo sv creator, use https://pkg.go.dev/github.com/sigstore/sigstore/pkg/signature#LoadSignerVerifier

Reply:
Sure! Thanks! I will try it out!